### DISCUSSION TITLE
Crazyflie 2.1 not working-- only M2 light works, able to enter bootloader but not flash or charge.

### DISCUSSION THREAD

**Discussion by abdnegm** [bitcraze/discussions#809] [Q&A]
After my crazyflie ran out of battery, I went to charge it, however I noticed that the M3 light that usually blinks blue did not work when charging it. Additionally, neither of the M1 or M4 lights worked. It is not able to charge, connect to radio, etc- only lights up M2 when drone is on, and flashes M2 when in bootloader mode (but can't flash new firmware). Appreciate any insights on next steps.

Thanks!

**Comment by ataffanel** [comment]
It is unlikely the battery is responsible for this situation.

It does look like the stm32 is not starting though and might need to be re-flashed. To do so, you need to use the bootloader [recovery mode](https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/userguides/recovery-mode/). To start both cpu in bootloader mode, you can:
 - Disconnect the battery
 - Press the ON/OFF button
 - Connect the battery while keeping the button pressed
 - Wait for the blue LED to blink and release the button

Both blue LEDs should now blink, you can use the client's recovery mode to flash the firmware.
